chalk-solve/src/coherence.rs

```diff
@@ -60,8 +61,7 @@ impl<I: Interner> SpecializationPriorities<I> {
     /// Store the priority of an impl (used during construction).
     /// Panics if we have already stored the priority for this impl.
     fn insert(&mut self, impl_id: ImplId<I>, p: SpecializationPriority) {
-        let old_value = self.map.insert(impl_id, p);
-        assert!(old_value.is_none());
```

Comment:
So, why is this assert removed?

Reply:
This assertion panics in https://github.com/rust-lang/chalk/commit/0c6d80689e4bf5cdb6015e968f9ba9c3d9482e5f. I have a feeling it is okay/correct to remove it.

Reply:
I don't really know this bit of code, so I'm weary of removing assertions without knowing why they were there, why they suddenly are failing, and why that's okay

Reply:
> I don't really know this bit of code, so I'm weary of removing assertions without knowing why they were there, why they suddenly are failing, and why that's okay

Agreed. I'll keep looking into this. Hopefully we can get to a good understanding of this behavior.